project/src/main/kotlin/org/ionproject/core/writeApi/insertClassSectionEvents/InsertClassSectionEventsController.kt

```diff
@@ -0,0 +1,106 @@
+package org.ionproject.core.writeApi.insertClassSectionEvents
+
+import com.fasterxml.jackson.databind.JsonNode
+import org.ionproject.core.common.Media
+import org.ionproject.core.common.ProblemJson
+import org.ionproject.core.join
+import org.ionproject.core.writeApi.common.Uri
+import org.ionproject.core.writeApi.insertClassSectionEvents.json.SchemaValidator
+import org.ionproject.core.writeApi.insertClassSectionEvents.json.InsertClassSectionEventsParams
+import org.jdbi.v3.core.statement.UnableToExecuteStatementException
+import org.postgresql.util.PSQLException
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+class InsertClassSectionEventsController(private val repo: InsertClassSectionEventsRepo) {
+
+  @PutMapping(Uri.insertClassSectionEvents, consumes = [Media.APPLICATION_JSON])
+  fun insertClassSectionEvents(@RequestBody json: JsonNode): ResponseEntity<Any> {
```

Comment:
It may be a personal opinion, but this method is gigantic, maybe try to refactor the sql interactions in smaller blocks?

Reply:
Will do :+1: